tests/pipeline/test_filter.py

```diff
@@ -395,6 +395,36 @@ def test_isfinite(self):
         )
         check_arrays(results['isfinite'], isfinite(data))
 
+    def test_smoothing_filter(self):
+        from zipline.pipeline.filters import SmoothingFilter
+
+        data = full(self.default_shape, True, dtype=bool)
+        # one column all false
+        data[0, 0] = False
+        data[1, 1] = False
+
+        class InputFilter(Filter):
+            inputs = ()
+            window_length = 0
+
+        smoothing_filter = SmoothingFilter(
+            inputs=[InputFilter()],
+            window_length=self.default_shape[0]
+        )
+
+        results = self.run_graph(
+            TermGraph({'smoothing': smoothing_filter}),
+            initial_workspace={InputFilter(): data}
+        )
+
+        expected_result = full(self.default_shape[1], True, dtype=bool)
+        expected_result[0] = False
+        expected_result[1] = False
+        check_arrays(
+            results['smoothing'].flatten(),
```

Comment:
This is losing information.  If `expected_result` was 4 x 4, and the actual output came back as 2 x 8, this wouldn't catch the error.  Instead of flattening here, you should `reshape` expected_result to the shape that's actually expected.